### ISSUE TITLE
header_rewrite: client-url:path and url:path don't work in string expansion

### ISSUE THREAD

**Issue by mlibbey** [apache/trafficserver#2989]
https://docs.trafficserver.apache.org/en/latest/admin-guide/plugins/header_rewrite.en.html#path
says PATH is deprecated as of ATS v7.1.x, please use e.g. `%{URL:PATH}` or
`%{CLIENT-URL:PATH}` instead.

However, when we try to use `%{URL:PATH}` or `%{CLIENT-URL:PATH}` in a string, they don't work, whereas `%{PATH}` does.

Reproduction:
a) Working/expected case
```
$ cat hdrs.config && curl -IXGET -x localhost:80 "http://httpbin.org/status/502" | grep Location
cond %{SEND_RESPONSE_HDR_HOOK}
cond %{STATUS} =502
  set-redirect 302 http://example.com/%{PATH} [QSA]

Location: http://example.com/status/502
```

b) using CLIENT-URL:PATH
```
$ cat hdrs.config && curl-IXGET -x localhost:80 "http://httpbin.org/status/502" | grep Location
cond %{SEND_RESPONSE_HDR_HOOK}
cond %{STATUS} =502
  set-redirect 302 http://example.com/%{CLIENT-URL:PATH} [QSA]

Location: http://example.com/%{CLIENT-URL:PATH}
```

c) using URL:PATH
```
$ cat hdrs.config && curl -IXGET -x localhost:80 "http://httpbin.org/status/502" | grep Location
cond %{SEND_RESPONSE_HDR_HOOK}
cond %{STATUS} =502
  set-redirect 302 http://example.com/%{URL:PATH} [QSA]

Location: http://example.com/%{URL:PATH}
```

**Comment by zwoop** [comment]
Hmmm, so as discussed, I don't see that e.g. %{PATH} should work in expansion either. What we really need to do is to rewrite the string expansions completely to:

1. Make it support all conditions as you'd expect (e.g. %{CLIENT-URL:PATH})
2. Remove the crazy log tags expansions (keeping those makes #1 uglier, unless we make operators for the log tags too)
3. Add an condition for a string literal (this becomes important for #4).
4. Optimize it such that at config parse time, it builds a list of the operators and strings that needs to be concatenated, instead of parsing the string every time it gets expanded.

#4 is a concern / request I made long ago but it was never implemented, and we landed the current implementation as-is :-/. But if we optimize this, a parsed expansion string would simply be a linked list of conditions, like

   ConditionString("Hello PATH=") -> ConditionURL(CLIENT-URL:PATH) -> ...

**Comment by randall** [comment]
I think I addressed this in #4568
